### DISCUSSION TITLE
Not Enough Space to Update

### DISCUSSION THREAD

**Discussion by whinson** [Hypfer/Valetudo#1676] [Q&A / Support] [answered]
Hi -

With having some free time, I am finally trying to upgrade my RoboRock S5 vacuum to the latest version.  I am really out of date, currently on version 2021.12.0.

When I use the updater through the UI, I get the following issue:

`Updating is impossible because there's not enough space to store the new binary at /usr/local/bin/valetudo. Required: 41943040 bytes. Available: 27691008 bytes.`  

In my mind, I see two potential paths forward:

1. SSH and delete things to free up space.  If this is the best approach forward, is there a quick answer on what to delete?
2. Create a new image on the latest version using DustBuilder and update that way.  If this is the best approach forward, are they are any additional things to consider?

Or is there another way forward that I am missing?

I see there was a similar question asked, but the discussion had been deleted and I can't find anything definitive in any of the documentation.

If anyone has any guidance or thoughts, I would greatly appreciate it.

Thank you in advance!

**Comment by Hypfer** [comment, marked as answer]
Why is your root partition this full? Did you install additional software?

During normal usage of the S5, that should not be happening.
I'd recommend the second approach. Just build a new image for manual install via ssh and install that

**Comment by whinson** [comment]
Thank you!  I will proceed with the new image.

To answer your question, it has been a long time since I have done anything with it, but I don't remember adding anything additional and not sure why I would have.

I've copied details about the filesystem and directory sizes below.  When I dig in, there is a resources folder and cleaner folder taking up 45 MB and 15 MB respectively.  And then a roborock file taking up 30 MB.  There is then a handful of smaller things taking up the rest of the use space.

If useful, I am happy to provide any other information.

```
Filesystem                Size      Used Available Use% Mounted on
rootfs                  169.5M    134.3M     26.4M  84% /
/dev/root               169.5M    134.3M     26.4M  84% /
devtmpfs                 64.0K         0     64.0K   0% /dev
none                    249.3M     20.0K    249.3M   0% /run
tmpfs                    30.0M    344.0K     29.7M   1% /var
tmpfs                    64.0K         0     64.0K   0% /dev
tmpfs                    30.0M      1.4M     28.6M   5% /tmp
tmpfs                   100.0M      2.5M     97.5M   3% /run/shm
/dev/mmcblk0p1            1.5G    127.0M      1.3G   9% /mnt/data
/dev/mmcblk0p6           62.0M     13.1M     45.7M  22% /mnt/default
/dev/mmcblk0p10         169.5M    134.6M     26.1M  84% /mnt/updbuf
/dev/mmcblk0p11          15.5M      1.6M     13.0M  11% /mnt/reserve

```


```
[root@rockrobo ]# du -sh * | sort -rn
590.0K  etc
344.0K  var
230.1M  mnt
63.1M   opt
48.2M   usr
12.0K   lost+found
10.1M   lib
3.7M    boot
3.0K    root
2.5M    run
1.6M    sbin
1.4M    tmp
1.4M    bin
1.0K    home
1.0K    build.txt
0       sys
0       proc
0       linuxrc
0       dev
```
